source/extensions/config_subscription/grpc/grpc_mux_impl.cc

```diff
@@ -520,5 +520,29 @@ void GrpcMuxImpl::drainRequests() {
   grpc_stream_.maybeUpdateQueueSizeStat(request_queue_->size());
 }
 
+class GrpcMuxFactory : public MuxFactory {
+public:
+  std::string name() const override { return "envoy.config_mux.grpc_mux_factory"; }
+  void shutdownAll() override { return GrpcMuxImpl::shutdownAll(); }
+  std::shared_ptr<GrpcMux>
+  create(Grpc::RawAsyncClientPtr&& async_client, Event::Dispatcher& dispatcher,
+         Random::RandomGenerator&, Stats::Scope& scope,
+         const envoy::config::core::v3::ApiConfigSource& ads_config,
+         const LocalInfo::LocalInfo& local_info, CustomConfigValidatorsPtr&& config_validators,
+         BackOffStrategyPtr&& backoff_strategy, XdsConfigTrackerOptRef xds_config_tracker,
+         XdsResourcesDelegateOptRef xds_resources_delegate) override {
+    return std::make_shared<Config::GrpcMuxImpl>(
+        local_info, std::move(async_client), dispatcher,
+        *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
+            "envoy.service.discovery.v3.AggregatedDiscoveryService.DeltaAggregatedResources"),
+        scope, Utility::parseRateLimitSettings(ads_config),
+        ads_config.set_node_on_first_message_only(), std::move(config_validators),
+        std::move(backoff_strategy), xds_config_tracker, xds_resources_delegate,
+        Config::Utility::getGrpcControlPlane(ads_config).value_or(""));
+  }
```

Comment:
LOL this is a 15 line method which is just a single make_shared() call. I love C++ :)